Command/SonataDoctrineUtilsCommand.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/*
+ * This file is part of the Sonata Connector project.
+ *
+ * (c) Sylvain Rascar <sylvain.rascar@fullsix.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Command;
+
+use Gaufrette\Filesystem;
+use Gaufrette\Adapter\Local as LocalAdapter;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Output\Output;
+use Psr\Log\InvalidArgumentException;
+
+/**
+ * Return useful data on the database schema
+ *
+ * Class SonataDoctrineUtils
+ * @package Sonata\ProductBundle\Command
+ */
+class SonataDoctrineUtilsCommand extends ContainerAwareCommand
+{
+    /**
+     * @var string
+     */
+    protected $defaultDumpDirectory;
+
+    /**
+     * @var array $allowedActions
+     */
+    protected $allowedActions = ['dump-meta', 'dump-entity-meta'];
+
+    /**
+     * @var array $metadata
+     */
+    protected $metadata;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('sonata:doctrine:utils')
+            ->setDefinition(
+                [
+                    new InputArgument(
+                        'action', InputArgument::REQUIRED, sprintf(
+                            'The action to execute [%s]',
+                            implode(' | ', array_keys($this->allowedActions))
+                        )
+                    ),
+                    new InputOption(
+                        'entity-name', 'e', InputOption::VALUE_OPTIONAL,
+                        'Name of the base entity to dump - ex: Product. Extended class will be dumped too.', null
+                    ),
+                    new InputOption(
+                        'filename', 'f', InputOption::VALUE_OPTIONAL,
+                        'If filename is specified, result will be dump into this file under json format.', null
+                    ),
+                ]
+            )
+            ->setDescription(
+                'Get information on the current Doctrine\'s schema'
+            );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->defaultDumpDirectory = $this->getContainer()->getParameter('sonata.doctrine.utils.dump_directory');
```

Comment:
It is already. It just prevent issues when using a wrong filename like "my_file.json"
(which would set directory to .)
but setting filename option to /my_directory/my_file.json works